mod: add kill assist + add team bleeding complaint, refs #2535

Aranud opened the pull request:
Kill assist a rewarded with Battle sense XP. 2 pts when more than half the max health has been removed, otherwise 1pts is granted. Only up to 4 players can be rewarded, selected by the highest amount of damage infliceted to the target.

g_teambleedComplaint has been added to control the minimum of damage required to max health to consider a team bleed has wanted. Negative value disable the feature.

Also add a new stats for highest assists in debriefing panels.

Aranud committed:
mod: add kill assist + add team bleeding complaint, refs #2535

Kill assist a rewarded with Battle sense XP. 2 pts when more than half the max health has been removed, otherwise 1pts is granted. Only up to 4 players can be rewarded, selected by the highest amount of damage infliceted to the target.

g_teambleedComplaint has been added to control the minimum of damage required to max health to consider a team bleed has wanted.
Negative value disable the feature.

Also add a new stats for highest assists in debriefing panels.

Aciz commented on src/cgame/cg_draw.c:
```diff
@@ -2732,8 +2734,8 @@ void CG_DrawVote(hudComponent_t *comp)
-		case -4: str = CG_TranslateString("You were team-killed by the Server Host"); break;
-		case -5: str = CG_TranslateString("You were team-killed by a bot."); break;
+		case -4: str = CG_TranslateString(va("You were %s by the Server Host", (cgs.complaintClient == CG_LastAttacker()) ? "team-killed" : "team-bleeded")); break;
+		case -5: str = CG_TranslateString(va("You were %s by a bot.", (cgs.complaintClient == CG_LastAttacker()) ? "team-killed" : "team-bleeded")); break;
```
`team-bleeded` isn't a word, it's `team-bled`

Aciz commented on src/game/g_stats.c:
```diff
@@ -574,6 +574,98 @@ void G_AddKillSkillPointsForDestruction(gentity_t *attacker, meansOfDeath_t mod,
+				}
+			}
+		}
+	}
```
I wonder if it would be simpler to have the `damageReceivedFromPlayers` array as a 2D array with `[damage][clientnum]`, if that was the case I believe this whole thing could just be a qsort?

Aranud committed:
cgame: fix team-bleeded isn't a word

Aranud committed:
cgame: display XP gain with decimal value

Also, fix right alignment

Aranud committed:
game: rework kill assist reward

Add a threshold of 1.5s since the last damage given to get the assist reward.
Rework XP gain depending of percentage of damage given to the assist (0 to 3 XP)
Use sort function to filter best damage given

Aranud committed:
game: format

Aranud committed:
mod: adjust assist header pos in score/debriefing

Aranud committed:
Merge branch 'master' into killassist

Aciz commented on src/game/g_local.h:
```diff
@@ -658,6 +658,7 @@ typedef struct
 	int spec_invite;
 	int spec_team;
 	int kills;
+	int kills_assists;
```
`kills_assists` to me implies this included both kills and assists, I'd just use either `kill_assists` or `assists` for this.
```suggestion
	int kill_assists;
```
```suggestion
	int assists;
```

Aciz commented on src/game/g_stats.c:
```diff
@@ -574,6 +574,98 @@ void G_AddKillSkillPointsForDestruction(gentity_t *attacker, meansOfDeath_t mod,
+		}
+
+		// reward only opposite team
+		if (ent->client->ps.teamNum != target->client->ps.teamNum)
```
This should be `sess.sessionTeam` for safety
```suggestion
		if (ent->client->sess.sessionTeam != target->client->sess.sessionTeam)
```

Aciz commented on src/cgame/cg_debriefing.c:
```diff
@@ -3948,6 +3970,7 @@ const char *awardNames[NUM_ENDGAME_AWARDS] =
 	"Highest Experience Points",
 	"Most Highly Decorated",
 	"Highest Fragger",
+	"Highest Assist",
```
`Most Assists` is more fitting here imo, just like you have `Most Deaths`.

```suggestion
	"Most Assists",
```

There's also perhaps a conversation to be had here whether this should just be the number of assists, or most damage via assists, as the assists reward XP based on the amount of damage dealt, but that would require quite a bit of rework, so perhaps something to revisit in the future after some playtesting and feedback.

Aciz reviewed (changes requested):
Seems to work fine, I left some suggestions that aren't really critical (except perhaps the team check in `G_AddKillAssisPoints`), more of opinionated comments.

One thing that needs to be fixed though is `G_LoseKillSkillPoints`, it will always say the reason for losing XP is team kill, even if the complaint was filed for team bleeding.

https://github.com/etlegacy/etlegacy/blob/455ffb8b190dbfeff2764fcf4cddae998d00eeab/src/game/g_stats.c#L485-L499

Aranud committed:
mod: rename / fix kill assist grammar var name

Aranud committed:
game: use sessTeam instead of teamNum

Aranud committed:
game: fix lost skill message for team bleeding

Aranud committed:
cgame: fix plurials ... words have meaning ...

Aranud committed:
mod: fix remaining kills_assists grammar
